packages/smart-contracts/test/contracts/BatchConversionPayments.test.ts

```diff
@@ -0,0 +1,840 @@
+    );
+
+    batchConversionProxy = await new BatchConversionPayments__factory(adminSigner).deploy(
+      erc20FeeProxy.address,
+      ethFeeProxy.address,
+      erc20ConversionProxy.address,
+      ethConversionProxy.address,
+      await adminSigner.getAddress(),
+    );
+
+    fauConvDetail.recipient = to;
+    daiConvDetail.recipient = to;
+    ethConvDetail.recipient = to;
+
+    // set batch proxy fees and connect fromSigner
+    await batchConversionProxy.setBatchFee(BATCH_FEE);
+    await batchConversionProxy.setBatchConversionFee(BATCH_CONV_FEE);
+    batchConversionProxy = batchConversionProxy.connect(fromSigner);
+
+    // set ERC20 tokens and transfer token to "from" (fromSigner)
+    daiERC20 = new TestERC20__factory(adminSigner).attach(DAI_address);
+    await daiERC20.transfer(from, BigNumber.from(thousandWith18Decimal + '0000000'));
+    daiERC20 = daiERC20.connect(fromSigner);
+
+    fauERC20 = new TestERC20__factory(adminSigner).attach(FAU_address);
+    await fauERC20.transfer(from, BigNumber.from(thousandWith18Decimal + '0000000'));
+    fauERC20 = fauERC20.connect(fromSigner);
+
+    await daiERC20.approve(batchConversionProxy.address, thousandWith18Decimal + fiatDecimals, {
+      from,
+    });
+    await fauERC20.approve(batchConversionProxy.address, thousandWith18Decimal + fiatDecimals, {
+      from,
+    });
+  });
+
+  const getERC20Balances = async (testERC20: TestERC20) => {
+    const fromDAIBalance = await testERC20.balanceOf(from);
+    const toDAIBalance = await testERC20.balanceOf(to);
+    const feeDAIBalance = await testERC20.balanceOf(feeAddress);
+    const batchDAIBalance = await testERC20.balanceOf(batchConversionProxy.address);
+    return [fromDAIBalance, toDAIBalance, feeDAIBalance, batchDAIBalance];
+  };
+
+  const getExpectedConvERC20Balances = (
+    amount: number,
+    fee: number,
+    nPayment: number,
+    path: string,
+  ) => {
+    // to get the exact result, we use millionDai
+    const conversionRate =
+      path === 'EUR_DAI'
+        ? BigNumber.from(millionDai).mul(120).div(101) // EUR_USD: 120, and DAI_USD: 101
+        : BigNumber.from(millionDai).mul(100).div(FAU_USD_RATE);
+    const expectedToDAIBalanceDiff = BigNumber.from(amount).mul(conversionRate).mul(nPayment);
+    const expectedFeeDAIBalanceDiff =
+      // fee added by the batch
+      expectedToDAIBalanceDiff
+        .add(BigNumber.from(fee).mul(conversionRate).mul(nPayment))
+        .mul(BATCH_CONV_FEE)
+        .div(BATCH_DENOMINATOR)
+        // fee within the invoice: .1% of the amount,
+        .add(BigNumber.from(fee).mul(conversionRate).mul(nPayment));
+    fee;
+    const expectedFromDAIBalanceDiff = expectedToDAIBalanceDiff
+      .add(expectedFeeDAIBalanceDiff)
+      .mul(-1);
+    return [
+      expectedFromDAIBalanceDiff.div('1000000'), // divide by 1 million because we used millionDai
+      expectedToDAIBalanceDiff.div('1000000'),
+      expectedFeeDAIBalanceDiff.div('1000000'),
+    ];
```

Comment:
```suggestion
    const precision = '1000000'; // Balance the non-support for decimals
    const EUR_USD = '120'; // TODO: move definition close to FAU_USD_RATE
    const DAI_USD = '101'; // TODO: move definition close to FAU_USD_RATE
    const conversionRate =
      path === 'EUR_DAI'
        ? BigNumber.from(daiDecimals).mul(precision).mul(EUR_USD).div(DAI_USD)
        : BigNumber.from(daiDecimals).mul(precision).mul(100).div(FAU_USD_RATE);
    const expectedToDAIBalanceDiff = BigNumber.from(amount).mul(conversionRate).mul(nPayment);
    const expectedFeeDAIBalanceDiff =
      // fee added by the batch
      expectedToDAIBalanceDiff
        .add(BigNumber.from(fee).mul(conversionRate).mul(nPayment))
        .mul(BATCH_CONV_FEE)
        .div(BATCH_DENOMINATOR)
        // fee within the invoice: .1% of the amount,
        .add(BigNumber.from(fee).mul(conversionRate).mul(nPayment));
    fee;
    const expectedFromDAIBalanceDiff = expectedToDAIBalanceDiff
      .add(expectedFeeDAIBalanceDiff)
      .mul(-1);
    return [
      expectedFromDAIBalanceDiff.div(precision),
      expectedToDAIBalanceDiff.div(precision),
      expectedFeeDAIBalanceDiff.div(precision),
    ];
```
This is a quick suggestion to replace required comments with explicit variable names.
But it looks very weird that we need to multiply by `daiDecimals` at the beginning, while we never devide by `daiDecimals`, everything should be denominated with the same level of precision, shouldn't it?

Reply:
The conversion aggregator multiplies by the daiDecimals.
If I delete daiDecimals, then I have to add it to the "amount" when I call this function.